pkg/kubernetes/pod/v1alpha1/build.go

```diff
@@ -115,6 +121,23 @@ func (b *Builder) WithNodeName(nodeName string) *Builder {
 	return b
 }
 
+// WithNodeSelectorHostname sets the Pod NodeSelector to the provided hostname value
+func (b *Builder) WithNodeSelectorHostname(hostname string) *Builder {
+	if len(hostname) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build Pod object: missing Pod hostname"),
+		)
+		return b
+	}
+
+	b.pod.object.Spec.NodeSelector = map[string]string{
+		k8sNodeLabelKeyHostname: hostname,
```

Comment:
We are resetting the node selector. It's not obvious to the callers.

Reply:
Hmm.. should it renamed as `WithNodeSelectorHostnameNew` or `WithNodeSelectorHostnameOnly`

Reply:
Discussed with amit and went ahead with `WithNodeSelectorHostnameNew`